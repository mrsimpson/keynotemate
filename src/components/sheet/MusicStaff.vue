<template>
  <div class="music-staff">
    <div class="staff-lines">
      <div class="line"></div>
      <div class="line"></div>
      <div class="line"></div>
      <div class="line"></div>
      <div class="line"></div>
    </div>
    <div class="notes-container">
      <slot></slot>
    </div>
  </div>
</template>

<style scoped>
.music-staff {
  position: relative;
  width: 100%;
  height: 120px;
  margin: 20px 0;
  display: flex;
}

.staff-lines {
  position: absolute;
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  padding: 20px 0;
}

.line {
  height: 1px;
  background-color: #ccc;
  width: 100%;
}

.notes-container {
  position: relative;
  flex-grow: 1;
  padding: 0 20px;
}
</style>
