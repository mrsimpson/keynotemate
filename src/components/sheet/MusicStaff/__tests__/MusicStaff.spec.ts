import { mount } from '@vue/test-utils'
import { describe, expect, it } from 'vitest'
import MusicStaff from '@/components/sheet/MusicStaff.vue'

describe('MusicStaff Component', () => {
  it('renders music staff snapshot', () => {
    const wrapper = mount(MusicStaff, {
      slots: {
        default: '<div class="test-note">Test Note</div>',
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders empty music staff snapshot', () => {
    const wrapper = mount(MusicStaff)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
