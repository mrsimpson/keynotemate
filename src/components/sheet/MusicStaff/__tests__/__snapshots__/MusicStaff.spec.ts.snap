// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MusicStaff Component > renders empty music staff snapshot 1`] = `
"<div data-v-faaa3425="" class="music-staff">
  <div class="staff-lines" data-v-faaa3425="">
    <div class="line" data-v-faaa3425=""></div>
    <div class="line" data-v-faaa3425=""></div>
    <div class="line" data-v-faaa3425=""></div>
    <div class="line" data-v-faaa3425=""></div>
    <div class="line" data-v-faaa3425=""></div>
  </div>
  <div data-v-faaa3425="" class="notes-container"></div>
</div>"
`;

exports[`MusicStaff Component > renders music staff snapshot 1`] = `
"<div data-v-faaa3425="" class="music-staff">
  <div class="staff-lines" data-v-faaa3425="">
    <div class="line" data-v-faaa3425=""></div>
    <div class="line" data-v-faaa3425=""></div>
    <div class="line" data-v-faaa3425=""></div>
    <div class="line" data-v-faaa3425=""></div>
    <div class="line" data-v-faaa3425=""></div>
  </div>
  <div data-v-faaa3425="" class="notes-container">
    <div class="test-note">Test Note</div>
  </div>
</div>"
`;
